### ISSUE TITLE
add an assist to convert a constructor parameter to a field (w/ initializer)

### ISSUE THREAD

**Issue by pq** [dart-lang/sdk#48970]
From flutter-insiders feedback:

> - When you extract a widget class, or add a new constructor parameter to an existing widget class via quick fix, what you end up in is this: 
> Constructor({String newVar});
> 
> However, below is where you want to get to and you have to edit the above manually to get to this below:
> Constructor({required this.newVar});
> String newVar;
> 
> It would be nice to have one or more refactor options that take us from the second one to the first one, such as "assign to a new field", "make required", etc.

**Comment by bwilkerson** [comment]
Note that there is work to consider for the existing quick fix(es), assuming we can identify which ones. Minimally, it should make the parameter `required` when in a null-safe library. But if there is anything that would indicate that a field is what the user is likely to want we should create the field directly without requiring a second step.

**Comment by FMorschel** [comment]
@bwilkerson, I think this has recently landed, right? I can't find the commit right now (I wasn't the one who implemented it, so I don't know which CL either; I just remember seeing it on the commit history some days ago).

**Comment by bwilkerson** [comment]
Yes, this was implemented in https://dart-review.googlesource.com/c/sdk/+/448202.
